src/test/java/com/school/project/controller/GroupControllerTest.java

```diff
@@ -0,0 +1,78 @@
+package com.school.project.controller;
+
+
+import com.school.project.model.entities.Group;
+import com.school.project.model.entities.Module;
+import com.school.project.service.GroupService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.Date;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+
+/*
+ * Unit test with dependency mocked.
+ */
+@RunWith(SpringRunner.class)
+@WebMvcTest(GroupController.class)
+@Import(ModelMapper.class)
+public class GroupControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private GroupService groupService;
+
+    @Test
+    public void testGetGroupById() throws Exception {
+        when(groupService.getGroupById(6L)).thenReturn(getSampleGroup());
+
+        mvc.perform(get("/group/" + 6)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andDo(print())//good for simple debugging
```

Comment:
as long as we don't have profiling, this line will print each run this test to console, what will do a lot of mess. So please don't use without purpose